src/main/java/TestRandomListGenerator.java

```diff
@@ -0,0 +1,12 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class TestRandomListGenerator {
+    public static void main(String[] args) throws NoSuchFieldException {
+        RandomGenerator rd = new RandomGenerator() ;
+
+        List<String> listOfString = new ArrayList<>(); // "it should be initialized like that to work"
```

Comment:
Not necessary. See how getRandomMap , and generateRandomObject works

Reply:
I implemented both versions by overloading

Reply:
forget about test file i pushed it by mistake